src/mococrw/hsm.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2022 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include <boost/optional/optional.hpp>
+#include "openssl_wrap.h"
+
+namespace mococrw
+{
+/**
+ * Driver for Hardware Security Modules (HSMs).
+ */
+class HSM
+{
+public:
+    HSM();
+    virtual ~HSM() {}
+
+protected:
+    /**
+     * Returns the name of the HSM driver.
+     */
+    virtual const std::string getName() = 0;
+
+    /**
+     *  Loads public key from HSM.
+     */
+    virtual openssl::SSL_EVP_PKEY_Ptr loadPublicKey(const std::string &keyID) = 0;
```

Comment:
What's the rational to return a openssl::SSL_EVP_PKEY_Ptr and not an object of type AsymmetricPublicKey or AsymmetricKeypair?

Usually this requires then an additional layer of abstraction to cast/transform it into the key object(s) used by MoCOCrW.

Reply:
This is an internal API which is to be used by AsymmetricPublicKey/AsymmetricKeypair (this is where the friend relationship comes in).